### PR TITLE
Model module #119

### LINKED ISSUE #119: Basic `Model` module
* only available to `admin` users
* basic handling of `relations`, `object_types`, `properties` and `property_types`

see https://docs.bedita.net/en/latest/endpoints/model/index.html

### PR DESCRIPTION
**Pull request by stefanorosanelli**
This PR resolves #119 
Save and delete actions have not been implemented for now.

 * a new `Model` module with routing rules, controller and views has been introduced
 * `roles` information added to user session to perform role-based checks -> `Model` module is only accessbile by `admin`
 * `moduleLink` variable is introduced: array with routing rules to create module main link

### CODE REVIEW

File: src/Controller/ModelController.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    /**
+     * {@inheritDoc}
+     */
+    public function initialize() : void
+    {
+        parent::initialize();
+
+        $this->loadComponent('Properties');
+
+        if (!empty($this->request)) {
```
Comment (batopa) [inline src/Controller/ModelController.php]: Can `$this->request` be  empty? It should always be populated `\Cake\Controller\Controller::__construct()`.
Maybe it should be:
```php
$resourceType = $this->request->getParam('resource_type');
if (!empty($resourceType)) {
    // etc...
}

```

File: src/Template/Pages/Model/index.twig [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                            {% elseif  propVal is same as(true) %}
+                                {% set propVal = 'Y' %}
+                            {% elseif  propVal is same as(false) %}
+                                {% set propVal = 'N' %}
+                            {% endif %}
+
+                            <div class="{{ prop }}-cell" untitled="{{ __('Untitled') }}">{{ propVal }}</div>
+                        {% endfor %}
+
+                        {# <div class="buttons">
```
Comment (batopa) [inline src/Template/Pages/Model/index.twig]: Is this comment here as a reminder? If yes I will write some note about why it is commented and/or what missing to make it works properly.

File: src/Template/Pages/Model/view.twig [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            </div>
+
+            <div class="form-column">
+                {% element 'Form/meta' {'cssOrder': 'bottom'}  %}
+            </div>
+
+        {{ Form.end()|raw }}
+
+        {# Append "Save" to commands in commands menu. Must stay here, before the form end. #}
+        {# {% do _view.append('commands', Form.submit(__('Save'), {'form': 'form-main'})) %} #}
```
Comment (batopa) [inline src/Template/Pages/Model/view.twig]: As above.

File: src/Template/Pages/Model/view.twig [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+        {# Append "Delete" to commands in commands menu. #}
+        {# {% do _view.append('commands', Form.postButton(__('Trash'), {'_name': 'modules:delete', 'resource_type': resourceType}, {'data': {'id': resource.id}})) %} #}
```
Comment (batopa) [inline src/Template/Pages/Model/view.twig]: As above

File: src/Controller/ModelController.php (hunk 1 above)
Comment (stefanorosanelli) [reply src/Controller/ModelController.php]: `$this->request` cannot be empty at this point, but Scrutinizer complains about that, maybe thers's come way to convince him :smiley_cat: 
Will remove the check and add `object_types`as default `resource_type` param.

File: src/Template/Pages/Model/index.twig (hunk 2 above)
Comment (stefanorosanelli) [reply src/Template/Pages/Model/index.twig]: Reminder to add buttons/actions to delete or change resources, not yet implemented... will add a comment

File: src/Template/Pages/Model/view.twig (hunk 3 above)
Comment (stefanorosanelli) [reply src/Template/Pages/Model/view.twig]: same as above response :wink:

File: src/Template/Pages/Model/view.twig (hunk 4 above)
Comment (stefanorosanelli) [reply src/Template/Pages/Model/view.twig]: same as above response :wink:

**Pull request merged**